spicy/toolchain/src/compiler/validator.cc

```diff
@@ -478,10 +479,22 @@ struct VisitorPost : visitor::PreOrder, hilti::validator::VisitorMixIn {
     }
 
     void operator()(spicy::declaration::Hook* n) final {
-        if ( auto field = n->parent<spicy::type::unit::item::Field>() ) {
-            if ( n->isForEach() && ! field->isContainer() )
-                error("'foreach' can only be used with containers", n);
+        int count = 0;
```

Comment:
An intermediary patch had rejection of attributes unsupported for hooks here which I think would be great to have (ideally with a test). With this I can write fields like this:

```ruby
    xs: uint8[] %error foreach &size=1 {}
```

Reply:
`foreach` and `%error` should be caught (except for your comment below about `count`, will check). `ip&v4` goes back to where we allow parsing attributes,  no? Seems that's a larger topic.